src/app/components/services/utilities/utility.service.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+/**
+ * @memberof transmartBaseUi
+ * @ngdoc factory
+ * @name UtilityService
+ * @description various utility functions
+ */
+angular.module('transmartBaseUi')
+    .factory('UtilityService', [function () {
+
+        var service = {};
+
+        /**
+         * Generate unique string
+         * @returns {string} unique string
+         * @memberof UtilityService
+         */
+        service.guid = function () {
+            function s4() {
+                return Math.floor((1 + Math.random()) * 0x10000)
+                    .toString(16)
+                    .substring(1);
+            }
+            return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
```

Comment:
And maybe take a look at this SO answer to generate a proper guid. It might not be that crucial in this case, but in case somebody else may rely on it in the future: http://stackoverflow.com/a/2117523/1097971